catalog/files/impl/src/test/java/org/projectnessie/catalog/files/s3/TestS3ProgrammaticOptions.java

```diff
@@ -0,0 +1,166 @@
+                    .region("region")
+                    .accessPoint("accessPoint")
+                    .allowCrossRegionAccessPoint(true)
+                    .serverAuthenticationMode(S3ServerAuthenticationMode.STATIC)
+                    .stsEndpoint(URI.create("https://stsEndpoint"))
+                    .assumeRole("assumeRole")
+                    .clientAuthenticationMode(S3ClientAuthenticationMode.REQUEST_SIGNING)
+                    .clientSessionDuration(Duration.ofSeconds(4))
+                    .externalId("externalId")
+                    .roleSessionName("roleSessionName")
+                    .sessionIamPolicy("sessionIamPolicy")
+                    .build())
+            .putBuckets(
+                "bucket1",
+                ImmutableS3NamedBucketOptions.builder()
+                    .name("bucket1")
+                    .endpoint(URI.create("https://host1"))
+                    .externalEndpoint(URI.create("https://externalHost1"))
+                    .pathStyleAccess(false)
+                    .region("region1")
+                    .accessPoint("accessPoint1")
+                    .allowCrossRegionAccessPoint(false)
+                    .serverAuthenticationMode(S3ServerAuthenticationMode.APPLICATION_GLOBAL)
+                    .stsEndpoint(URI.create("https://stsEndpoint1"))
+                    .assumeRole("assumeRole1")
+                    .clientAuthenticationMode(S3ClientAuthenticationMode.REQUEST_SIGNING)
+                    .clientSessionDuration(Duration.ofSeconds(5))
+                    .externalId("externalId1")
+                    .roleSessionName("roleSessionName1")
+                    .sessionIamPolicy("sessionIamPolicy1")
+                    .build())
+            .putBuckets(
+                "my-bucket-2",
+                ImmutableS3NamedBucketOptions.builder()
+                    .name("my-bucket-2")
+                    .endpoint(URI.create("https://host2"))
+                    .externalEndpoint(URI.create("https://externalHost2"))
+                    .pathStyleAccess(true)
+                    .region("region2")
+                    .accessPoint("accessPoint2")
+                    .allowCrossRegionAccessPoint(true)
+                    .serverAuthenticationMode(S3ServerAuthenticationMode.STATIC)
+                    .stsEndpoint(URI.create("https://stsEndpoint2"))
+                    .assumeRole("assumeRole2")
+                    .clientAuthenticationMode(S3ClientAuthenticationMode.REQUEST_SIGNING)
+                    .clientSessionDuration(Duration.ofSeconds(6))
+                    .externalId("externalId2")
+                    .roleSessionName("roleSessionName2")
+                    .sessionIamPolicy("sessionIamPolicy2")
+                    .build())
+            .build();
+    S3ProgrammaticOptions actual = S3ProgrammaticOptions.copyOf(input);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  void normalizeBuckets() {
```

Comment:
Would be good to test the good case as well, and that the bucket options are contained using the `.name()` in S3/GCS/ADLSOptions.